### PR TITLE
Improve workflow config

### PR DESCRIPTION
**Pull request by patcon**
This PR:
- Locked workflows' Node versions using `admin/.nvmrc` (arbitrarily chose the sub-project seemingly most active)
- Locked workflows' NPM versions to 8.3.0 (instead of floating with `latest`, so CI will only change when we change it)
- Only run storybook workflow when appropriate files change
- Only run storybook workflow on mainline pushes, and PRs

### CODE REVIEW

**Review by petertgiles — APPROVED** [review]
This all looks really good to me.  I'll let Tristan do the merge since this is "github stuff" and he's the repo owner.
Regarding `Locked workflows' Node versions using admin/.nvmrc ` this seems reasonable for now.  Once the frontend projects are merged we should have a more meaningful rc file to use.

**Review by tristan-orourke — APPROVED** [review]
Thanks, looks good

**Pull request merged**
